### PR TITLE
WEBAPP-112: Refactored folder structure following this guide: https://engineering.opsgenie.com/how-to-organize-react-files-before-its-messed-up-c85387f691be

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

**Comment by maxammann** [comment]
@klinzo 

1. A route or module can have multiple containers. Therefore I would stick to the containers package
2. "A container does data fetching and then renders its corresponding sub-component."
So yes Breadcrump shouldn't be a container but a component. This needs further refactoring and can be done when testing the containers.
3. 👍 
4. I would pefere that there are no more nested components which are declared inside other components. All components should be declared in components/ and containers/. So you know if a are looking for a component by name you can just look in the folder and will find it on the first layer.
5. Layout is dump, RickLayout fetches data from state.
6. As our app gets more complicated we will have more modules like endpoint. The app module contains all components which are essential for the structure of the app. And the App component is one of those.
7. 👍 for content

**Comment by Schedulaar** [comment]
Since you're currently working on it - you could apply the naming-changes from my WEBAPP-112 PR, like categoryTiles or SearchInput, etc.
Full review is following...

**Comment by maxammann** [comment]
@Schedulaar renaming is ok, but I'd move the Hierarchy url change to a separate PR.

**Comment by klinzo** [comment]
@maxammann 
1) i think you misunderstood me, i didn't mean we don't need a container package in the different route, but no extra package LocationPage or EventsPage and so on in this container packages, as each route can only have one "root" container which is our LocationPage, EventsPage and so on, so imo this could just be moved into the container package and being renamed to index.js and style.css
2) ok perfect, should I create a task?
4) I agree, this was what i meant
5) imo it should be a container anywhere as it contains other components, which is the usual meaning of the word container
6) agreed
7) was also my first pick

**Comment by maxammann** [comment]
@klinzo 
1. This would conflict with the thumb-rule that all component file should be Uppercase or be in a package which is upper-case
2. We have to create a task to do this for multiple containers so let's do this after this PR
4. Apparently not in the react world :P https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

**Comment by klinzo** [comment]
@maxammann 
1) i don't like it anyway, as its just a thumb rule couldn't we make an exception here?
2) okay
3) alright you convinced me :D

**Comment by maxammann** [comment]
@klinzo 
1. Let's see what @Schedulaar thinks.
If we have a `index.js` inside a folder then the actual name of the file is the name of that folder. So we would have here a file named `containers.js`. Maybe we make it singular so: `container.js`. Inside that file is a class named e.g. `ContentPage`. All other components follow an other naming rule. So I think we should stick to it.

**Comment by klinzo** [comment]
@maxammann 
1) okay sorry you're right, i hadn't had in mind that there would be 6 container files or so, so maybe moving index.js and style.css of the subfolder LocationPage directly in container but named as LocationPage.js and LocationPage.css would be ok?

**Comment by maxammann** [comment]
@klinzo 
1. So far that was ok for me. Not sure about the test files, though:
I like this default folder structure: 
```
FilterableLocationSelector
├── index.js
├── style.css
└── __tests__
    ├── index.js
    └── __snapshots__
        └── index.js.snap
```

The question is: What would happen to __tests__.

**Comment by maxammann** [comment]
We could get rid of the style.css with this: https://www.styled-components.com/

**Comment by klinzo** [comment]
@maxammann 
regarding getting rid of the style.css, do we need to get rid of it?

**Comment by maxammann** [comment]
@klinzo not really, styled-components is just a step close to react-native :)
Alternative component structure:

```
components
├── FilterableLocationSelector.js
├── FilterableLocationSelector.css
├── OtherComponent.js
├── OtherComponent.css
├── Comp.css
└── __tests__
    ├── FilterableLocationSelector.js
    ├── OtherComponent.js
    ├── Comp.js
    └── __snapshots__
        └── FilterableLocationSelector.js.snap
```

**Comment by klinzo** [comment]
@maxammann 
i like the second suggestions way better because otherwise we have lots of folders only containing one js and one css file

File: src/routes/content/containers/CategoryTiles.js [hunk 1]
```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types'
...
-import Caption from './Caption'
+import PageModel from 'modules/endpoint/models/PageModel'
+import LOCATIONS_ENDPOINT from 'modules/endpoint/endpoints/location'
+import withFetcher from 'modules/endpoint/hocs/withFetcher'
+import Caption from 'modules/common/components/Caption'
 
-import style from './TitledCategoriesTable.css'
+import style from './CategoryTiles.css'
 
 class Category extends React.Component {
```
Comment (Schedulaar) [inline src/routes/content/containers/CategoryTiles.js]: Rename to CategoryTile

File: www/index.ejs [hunk 2]
```diff
@@ -72,7 +72,7 @@
   </style>
 </head>
 <body>
-<div id="splash" class="splash"><img class="splash-logo" src="/progressive-logo.jpg"></div>
+<div id="splash" class="splash"><img class="splash-logo" src="/www/progressive-logo.jpg.jpg"></div>
```
[the remark this answers is not in the document]
Comment (maxammann) [reply www/index.ejs]: Done

File: www/index.ejs [hunk 3]
```diff
@@ -27,8 +27,8 @@
 
 
 
-  <link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
-  <link rel="apple-touch-icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
+  <link rel="icon" type="image/png" sizes="96x96" href="/www/favicon-96x96.png">
```
[the remark this answers is not in the document]
Comment (maxammann) [reply www/index.ejs]: I agree, sry :/

File: www/index.ejs [hunk 4]
```diff
@@ -6,12 +6,12 @@
...
-  <link rel="apple-touch-icon" href="/apple-touch-icon.png">
+  <link rel="apple-touch-icon" href="/www/apple-touch-icon.png">
 
-  <link rel="icon" type="image/png" href="/favicon-96x96.png" sizes="96x96">
-  <link rel="icon" type="image/png" href="/favicon-16x16.png" sizes="16x16">
+  <link rel="icon" type="image/png" href="/www/favicon-96x96.png" sizes="96x96">
+  <link rel="icon" type="image/png" href="/www/favicon-16x16.png" sizes="16x16">
 
-  <link rel="manifest" href="/manifest.json">
+  <link rel="manifest" href="/www/manifest.jsonjson">
```
[the remark this answers is not in the document]
Comment (maxammann) [reply www/index.ejs]: Done

File: www/index.ejs [hunk 5]
```diff
@@ -6,12 +6,12 @@
   <title><%= config.title %></title>
   <meta name="description" content="">
   <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="apple-touch-icon" href="/apple-touch-icon.png">
+  <link rel="apple-touch-icon" href="/www/apple-touch-icon.png">
```
[the remark this answers is not in the document]
Comment (maxammann) [reply www/index.ejs]: Done

File: src/routes/events/components/EventList.css [hunk 6]
```diff
@@ -1,5 +1,5 @@
-@import url('../Common/Helper.css');
-@import url('../Common/Colors.css');
+@import url('../../../modules/common/constants/helpers.css');
+@import url('../../../modules/common/constants/colors.css');
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/routes/events/components/EventList.css]: I tried it for half an hour but couln't get postcss to use find the stylesheets :/

[the remark this answers is not in the document]
Comment (Schedulaar) [reply src/routes/events/components/EventList.css]: I remembered trying it as well 🤔

**Comment by maxammann** [comment]
@Schedulaar 
1. I enforced the following naming convention now: 'rrrr-ssss' is in 'routes/rrrr/containers/RrrrSsssPage.js'
2. Yes, thats how we're going to change this. I created tasks for every module/route to refactor and test. We'll split containers and components as good as possible in those tasks.
7. How about just categories? Like it is in our UI

**Comment by Schedulaar** [comment]
How about having
```
├── components
├── containers
└── utils
    └── Hierarchy.js
```
?

**Comment by maxammann** [comment]
@Schedulaar hm makes sense in this case but we don't really have many utils. We have many files in the endpoint module. What should we do with them?

If we have the case that we have many util I think we should think about making a new module instead of having module-local utils.

**Comment by Schedulaar** [comment]
@maxammann 
Yoah, I think you're right. 👍

**Pull request closed without merging**